proposals/low-level-struct-improvements.md

```diff
@@ -247,11 +246,11 @@ Other uses for `scoped` on locals are discussed [below](#examples-scoped-locals)
 
 When `scoped` is applied to an instance method the `this` parameter will have the type `scoped ref T`. By default this is redundant as `scoped ref T` is the default type of `this`. It is useful in the case the `struct` is declared as `unscoped` (detailed [below](#return-fields-by-ref)).
 
-The `scoped` annotation cannot be applied to any other location including returns, fields, array elements, etc ... Further while `scoped` can be applied to any `ref`, `in` or `out` it can only be applied to values which are `ref struct`. Having declarations like `scoped int` adds no value to the rules and will be prevented to avoid developer confusion.
+The `scoped` annotation cannot be applied to any other location including returns, fields, array elements, etc ... Further while `scoped` has impact when applied to any `ref`, `in` or `out` it only has impact when applied to values which are `ref struct`. Having declarations like `scoped int` has no impact because a non `ref struct` is always safe to return. The compiler will create a diagnostic on such cases to avoid developer confusion.
```

Comment:
>The compiler will create a diagnostic on such cases to avoid developer confusion. [](http://example.com/codeflow?start=361&length=81)

We could be more specific: It is an error to use `scoped` for a value that isn't a `ref` or a `ref struct`.

Reply:
Made it more specific but chose warning over error. The reasoning is that making it an error places burden on generator tools. They have to do semantic analysis before knowing if a type is or is not `ref struct` before they can emit `scoped` on the parameter. Historically in LDM we've preferred warning in these scenarios to benefit generators. Basically it's not wrong, it's just not doing what you want.

Reply:
I'm pointing out that the current implementation. So either the spec can say error, or the implementation should be discussed/adjusted.